debian/patches/0069-add-tonemap-videotoolbox.patch

```diff
@@ -0,0 +1,1867 @@
++        [command_buffer waitUntilCompleted];
++    }
++
++    ctx->mtl_function = [ctx->mtl_library newFunctionWithName:@"tonemap" constantValues:constant_values error:&ns_error];
++    if (ns_error) {
++        av_log(ctx, AV_LOG_ERROR, "Failed to create Metal function: %s\n", ns_error.description.UTF8String);
++        err = AVERROR_EXTERNAL;
++        goto fail;
++    }
++
++    ctx->mtl_pipeline = [ctx->mtl_device newComputePipelineStateWithFunction:ctx->mtl_function error:&ns_error];
++    if (ns_error) {
++        av_log(ctx, AV_LOG_ERROR, "Failed to create Metal compute pipeline: %s\n", ns_error.description.UTF8String);
++        err = AVERROR_EXTERNAL;
++        goto fail;
++    }
++
++    if (dovi_reshape) {
++        ctx->mtl_dovi_buffer = [ctx->mtl_device newBufferWithLength: 3*(params_sz+pivots_sz+coeffs_sz+mmr_sz)
++                                                            options: MTLResourceStorageModeShared];
++        if (!ctx->mtl_dovi_buffer) {
++            av_log(ctx, AV_LOG_ERROR, "Failed to create Metal buffer for Dolby Vision data\n");
++            err = AVERROR_EXTERNAL;
++            goto fail;
++        }
++    }
++
++    ctx->mtl_peak_buffer = [ctx->mtl_device newBufferWithLength: sizeof(float)
++                                                        options: MTLResourceStorageModeShared];
++    if (!ctx->mtl_peak_buffer) {
++        av_log(ctx, AV_LOG_ERROR, "Failed to create Metal buffer for Peak data\n");
++        err = AVERROR_EXTERNAL;
++        goto fail;
++    }
++
++    ctx->initialised = 1;
++    return 0;
++
++fail:
++    tonemap_videotoolbox_uninit(avctx);
++    return err;
++}
++
++static int tonemap_videotoolbox_config_output(AVFilterLink *outlink)
++{
++    AVFilterContext    *avctx = outlink->src;
++    AVFilterLink      *inlink = avctx->inputs[0];
++    TonemapVideoToolboxContext *ctx = avctx->priv;
++    AVHWFramesContext *in_frames_ctx, *out_frames_ctx;
++    enum AVPixelFormat in_format;
++    enum AVPixelFormat out_format;
++    const AVPixFmtDescriptor *in_desc;
++    const AVPixFmtDescriptor *out_desc;
++    int ret;
++
++    if (!inlink->hw_frames_ctx)
++        return AVERROR(EINVAL);
++    in_frames_ctx = (AVHWFramesContext*)inlink->hw_frames_ctx->data;
++    in_format     = in_frames_ctx->sw_format;
++    out_format    = (ctx->format == AV_PIX_FMT_NONE) ? in_format : ctx->format;
++    in_desc       = av_pix_fmt_desc_get(in_format);
++    out_desc      = av_pix_fmt_desc_get(out_format);
++
++    if (!format_is_supported(in_format)) {
++        av_log(ctx, AV_LOG_ERROR, "Unsupported input format: %s\n",
++               av_get_pix_fmt_name(in_format));
++        return AVERROR(ENOSYS);
++    }
++    if (!format_is_supported(out_format)) {
++        av_log(ctx, AV_LOG_ERROR, "Unsupported output format: %s\n",
++               av_get_pix_fmt_name(out_format));
++        return AVERROR(ENOSYS);
++    }
++    if (in_desc->comp[0].depth != 10 && in_desc->comp[0].depth != 16) {
```

Comment:
P016?

Reply:
As mentioned above, this is a "just-in-case" option. I don't have any video with this format to test, so I disabled it in the `supported_formats` section, but the code path is kept.